src/main/java/io/cdap/e2e/pages/actions/WranglerActions.java

```diff
@@ -0,0 +1,260 @@
+        selectDropdown.selectByVisibleText(option);
+        ElementHelper.clickOnElementUsingJsExecutor(WranglerLocators.selectDirectiveOption(option));
+        break;
+
+      case "Datetime":
+      case "Simple date":
+        if (option.equals("Custom format")) {
+          ElementHelper.clickOnElement(WranglerLocators.selectDirectiveOption(directiveType));
+          ElementHelper.clickOnElement(WranglerLocators.selectRadioButton(option));
+          ElementHelper.sendKeys(WranglerLocators.customDate, text);
+        }
+        break;
+      case "Always":
+        Select select = new Select(WranglerLocators.filterSelect);
+        select.selectByVisibleText(directiveType);
+        ElementHelper.sendKeysToTextarea(WranglerLocators.enterCounterName, text);
+        ElementHelper.replaceElementValue(WranglerLocators.incrementCount, option);
+        break;
+    }
+    ElementHelper.clickOnElement(WranglerLocators.applyButton);
+  }
+
+  public static void selectDirectiveTypeWithFiveOption(String columnName, String directive, String option1,
+                                                       String option2, String option3, String option4)
+    throws InterruptedException {
+    ElementHelper.clickOnElement(WranglerLocators.locateTransformationButton(columnName));
+    String pluginPropertyDirective = PluginPropertyUtils.getPluginPropertyElementTestId(directive);
+    ElementHelper.clickOnElement(WranglerLocators.locateDirectivesTitle(pluginPropertyDirective));
+    switch (directive) {
+      case "SetCounter":
+        Select select = new Select(WranglerLocators.filterSelect);
+        select.selectByVisibleText(option1);
+        ElementHelper.sendKeysToTextarea(WranglerLocators.filterTextArea, option2);
+        ElementHelper.sendKeysToTextarea(WranglerLocators.enterCounterName, option4);
+        ElementHelper.replaceElementValue(WranglerLocators.incrementCount, option3);
+        break;
+
+      case "DefineVariable":
+        ElementHelper.sendKeysToTextarea(WranglerLocators.variableName, option4);
+        Select selectColumnName = new Select(WranglerLocators.selectColumn);
+        selectColumnName.selectByVisibleText(option3);
+        Select selectVar = new Select(WranglerLocators.selectRow);
+        selectVar.selectByVisibleText(option1);
+        if (option1.equals("Custom condition")) {
+          ElementHelper.sendKeysToTextarea(WranglerLocators.filterTextArea, option2);
+        } else {
+          ElementHelper.sendKeysToTextarea(WranglerLocators.variableValue, option2);
+        }
+        break;
+    }
+    ElementHelper.clickOnElement(WranglerLocators.applyButton);
+  }
+
+
+  public static void enterDirectiveFromCommandLine(String directive) throws InterruptedException {
+    ElementHelper.sendKeys(WranglerLocators.directiveCommandLine, directive);
+    Actions act = new Actions(SeleniumDriver.getDriver());
+    act.sendKeys(new CharSequence[]{Keys.ENTER}).perform();
+    Thread.sleep(2000);
```

Comment:
Remove static sleep statements and replace it with dynamic ones. We should know what we are waiting for and for how long.

Reply:
done